src/extractor/restriction_parser.cpp

```diff
@@ -54,9 +56,10 @@ RestrictionParser::RestrictionParser(ScriptingEnvironment &scripting_environment
  * in the corresponding profile. We use it for both namespacing restrictions, as in
  * restriction:motorcar as well as whitelisting if its in except:motorcar.
  */
-boost::optional<InputRestrictionContainer>
+std::vector<boost::optional<InputRestrictionContainer>>
```

Comment:
We can get rid of the `boost::optional` here. The `optional` was only used because a function needs to return something even if the parsing fails. In this case we can just return an empty vector. 🎉